File: trainer-flutter/lib/config/api_config.dart
// ignore: avoid_classes_with_only_static_members
class ApiConfig {
  /// Railway NestJS backend
  static const String baseUrl =
      'https://admin-backend-php-production.up.railway.app/api/';

  static const String salt = r'sKLUIE7dfwo4hn23l;idfj[028325p*^&)(op';
  static const String authHeader = 'X-Auth-Token';

  // Auth & User
  static const String trainerMe         = 'trainer/me';

  // Studio info  (PHP-only — returns 404 on NestJS, handled gracefully)
  static const String aboutUs           = 'trainer/aboutus';

  // Locations
  static const String locationList      = 'location';

  // Clients
  static const String client            = 'client';

  // Trainer availability
  static const String availability      = 'trainer-availability';
  static const String availabilitySerial = 'trainer-availability/serial';

  // Trainings / Appointments
  static const String training          = 'training';

  // Feedback (star ratings from clients)
  static const String feedback          = 'feedback';

  // Training plans
  static const String trainingPlan      = 'training-plan';

  // Performance / metrics
  static const String metric            = 'metric';

  // Performance tests
  static const String performance       = 'performance_test';

  // Anamnese (medical intake questionnaire)
  static const String anamnese          = 'anamnese';

  // Reviews (training recordings with heart rate)
  static const String review            = 'review';

  // ── PHP-legacy endpoints (not yet in NestJS — will return 404) ─────────────
  // These are referenced in screens but not yet implemented in NestJS.
  // They fail gracefully with an error message shown to the user.
  static const String file              = 'file';
  static const String sendFile          = 'file/send';
  static const String trainerQR         = 'trainer/qr';
  static const String preference        = 'preference';
  static const String changePassword    = 'trainer/password';
}

File: trainer-ios/SihlTrainer/Config/APIConfig.swift
import Foundation

/// API-Konfiguration — zeigt auf dasselbe Railway-NestJS-Backend wie die
/// Flutter-Trainer-App und die Client-App.
/// Pendant zu `trainer-flutter/lib/config/api_config.dart`.
enum APIConfig {
    static let baseURL = URL(string: "https://admin-backend-php-production.up.railway.app/api/")!

    /// Auth-Header-Name (token-basierte Auth)
    static let authHeader = "X-Auth-Token"

    /// Request-Timeout
    static let timeout: TimeInterval = 30

    /// Salt des Legacy-PHP-Auth-Schemas: Token = md5(salt + passcode).
    /// Steht so auch im Flutter-Client — in der Web-App liegt er im
    /// ausgelieferten JS-Bundle offen. Die Sicherheit hängt allein an der
    /// Passcode-Entropie; das Schema wandert mit, damit die App gegen das
    /// bestehende Backend funktioniert.
    static let salt = #"sKLUIE7dfwo4hn23l;idfj[028325p*^&)(op"#

    // MARK: - Endpunkte (Reihenfolge wie in api_config.dart)

    static let trainerMe          = "trainer/me"
    static let aboutUs            = "trainer/aboutus"
    static let locationList       = "location"
    static let client             = "client"
    static let availability       = "trainer-availability"
    static let availabilitySerial = "trainer-availability/serial"
    static let training           = "training"
    static let feedback           = "feedback"
    static let trainingPlan       = "training-plan"
    static let exercise           = "exercise"
    static let exerciseGroups     = "exercise/groups"
    static let metric             = "metric"
    static let performance        = "performance_test"
    static let anamnese           = "anamnese"
    static let review             = "review"
    static let file               = "file"
    static let sendFile           = "file/send"
    static let trainerQR          = "trainer/qr"
    static let preference         = "preference"
    static let changePassword     = "trainer/password"
}
